src/server/routes/import-entity/author.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2018 Shivam Tripathi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as auth from '../../helpers/auth';
+import * as importRoutes from './import-routes';
+import * as middleware from '../../helpers/middleware';
+import * as utils from '../../helpers/utils';
+import express from 'express';
+
+
+const router = express.Router();
+
+/* If the route specifies an importId, load the importEntity for it. */
+router.param(
+	'importId',
+	middleware.makeImportLoader(
+		'AuthorImport',
+		['authorType', 'gender', 'beginArea', 'endArea'],
+		'Author Import not found'
+	)
+);
+
+function _setAuthorTitle(res) {
+	res.locals.title = utils.createEntityPageTitle(
+		res.locals.entity,
+		'Author (Import)',
+		utils.template`Author (Import) “${'name'}”`
+	);
+}
+
+router.get('/:importId', (req, res) => {
+	_setAuthorTitle(res);
+	importRoutes.displayImport(req, res);
+});
+
+router.get('/:importId/discard', auth.isAuthenticated, (req, res) => {
+	_setAuthorTitle(res);
+	importRoutes.displayDiscardImportEntity(req, res);
+});
+
+router.post(
+	'/:importId/discard/handler',
+	auth.isAuthenticatedForHandler,
+	importRoutes.handleDiscardImportEntity
+);
+
+router.get(
+	'/:importId/approve',
+	auth.isAuthenticatedForHandler,
+	importRoutes.approveImportEntity
+);
+
+router.get(
+	'/:importId/edit',
+	middleware.loadIdentifierTypes, middleware.loadGenders,
+	middleware.loadLanguages, middleware.loadAuthorTypes,
+	importRoutes.editImportEntity
```

Comment:
For some entity types, we need to load more information and add some of it to the page props.
In this case for example, `loadGenders` loads the available genders from the DB into `res.locals.genders`, and needs to be extracted from there and passed into page props like we do for the regular entities edit pages:
https://github.com/metabrainz/bookbrainz-site/blob/7c23ed6a782a1e6aaef03b5cec82a490da902de6/src/server/routes/entity/author.ts#L340-L344

Currently the `editImportEntity` function does not allow that, but maybe it can be modified to allow attaching certain extras from res.locals to props, something like:
```
if(res.locals.genders){
  optionalProps.genderOptions = res.locals.genders
}
```

Reply:
I will look into this next, so far I haven't even tried the approve/edit action buttons, only the discard button which leads to a confirmation page.